File: Analisis_data_shopee/shopee/4.py
import pandas as pd
import mysql.connector
import numpy as np
import matplotlib.pyplot as plt

connection = mysql.connector.connect(host='localhost', user='root', password='', database='pembelian_shopee')
query1 = "SELECT*FROM order_detail"
query2 = "SELECT*FROM payment_detail"
df1 = pd.read_sql(query1, connection)
df2 = pd.read_sql(query2, connection)
connection.close()

df = pd.merge(df1, df2, left_on='payment_id', right_on='id', how='inner')
df = df[df['is_valid'] == 1]
df['order_date'] = pd.to_datetime(df['order_date'])
df = df[df['order_date'].dt.year == 2022]
print(df)
hasil = df.groupby('payment_method')['id_x'].nunique().reset_index(name='total').sort_values('total', ascending=False).reset_index(drop=True)
print(hasil)
print(hasil.head(5))

File: Analisis_data_shopee/shopee/4.r
library(RMySQL)
library(dplyr)
library(tidyr)
library(lubridate)

# Ganti 'dbname', 'username', 'password', dan 'host' sesuai dengan informasi database Anda
con <- dbConnect(MySQL(), dbname = "pembelian_shopee", user = "root", password = "", host = "localhost")

query1 <- "SELECT * FROM order_detail"
query2 <- "SELECT * FROM payment_detail"

df1 <- dbGetQuery(con, query1)
df2 <- dbGetQuery(con, query2)
dbDisconnect(con)

print(head(df1))
cat("\n")
print(head(df2))
cat("\n\n")


df <- merge(df1, df2, by.x = "payment_id", by.y = "id", all = FALSE)
print(head(df))

df <- filter(df, is_valid == 1)
df$order_date <- as.Date(df$order_date)
df <- filter(df, year(order_date) == 2022)
print(head(df))

hasil <- df %>% group_by(payment_method) %>% summarise(total = n_distinct(id)) %>% arrange(desc(total))
print(hasil)
print(head(hasil, 5))